### PR TITLE
tiflash: add tiflash tuning document

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add TiFlash performance tuning document.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2603 https://github.com/pingcap/docs-cn/pull/2642 https://github.com/pingcap/docs-cn/pull/2686
- Other reference link(s):

### CODE REVIEW

**Comment by yikeke** [comment]
Please help review this PR. @ran-huang

File: reference/tiflash/tune-performance.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+For user scenarios that focus on analysis and query, it is recommended that you set the Region size to 192MB, and enable the Region Merge feature to ensure that small Regions generated during operation are automatically merged.
+
+This section introduces how to set Region size and enable Region Merge for TiDB clusters that are deployed in different ways.
+
+### For clusters deployed using TiDB Ansible
+
+For clusters deployed using TiDB Ansible, use the following methods to set Region size and enable Region Merge:
+
++ Add the following configuration in the `roles/tikv/vars/default.yml` file and then restart the cluster to modify the Region size.
```
Comment (ran-huang) [inline reference/tiflash/tune-performance.md:45]:
```suggestion
+ To modify the Region size, add the following configuration in the `roles/tikv/vars/default.yml` file and then restart the cluster.
```

File: reference/tiflash/tune-performance.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    ```yaml
+    coprocessor:
+    region-max-size: "288MiB"
+    region-split-size: "192MiB"
+    region-max-keys:  2880000
+    region-split-keys:  1920000
+    ```
+
++ Add the following configuration in the `roles/pd/vars/default.yml` file to enable Region Merge. You can also execute the `config set <config-name>` command in pd-ctl to enable this feature.
```
Comment (ran-huang) [inline reference/tiflash/tune-performance.md:57]:
```suggestion
+ To enable Region Merge, add the following configuration in the `roles/pd/vars/default.yml` file. You can also execute the `config set <config-name>` command in pd-ctl to enable this feature.
```

File: reference/tiflash/tune-performance.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    {{< copyable "" >}}
+
+    ```yaml
+    schedule:
+    max-merge-region-size: 20 # Merges Regions smaller that 20MB.
```
Comment (ran-huang) [inline reference/tiflash/tune-performance.md:63]:
```suggestion
    max-merge-region-size: 20 # Merges Regions smaller that 20 MB.
```

File: reference/tiflash/tune-performance.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    max-merge-region-keys: 200000 # Merges Regions with less than 20,000 keys.
+    split-merge-interval: "1h"
+    merge-schedule-limit: 8 # Controls the speed of merge. Setting this value to 0 will disable Region Merge.
+    ```
+
+### For clusters deployed not using TiDB Ansible
+
+For TiDB clusters deployed not using TiDB Ansible, use the following methods to set Region size and enable Region Merge:
+
++ Add the following configuration in the `tikv.toml` file and then restart the cluster to modify the Region size.
```
Comment (ran-huang) [inline reference/tiflash/tune-performance.md:73]:
```suggestion
+ To modify the Region size, add the following configuration in the `tikv.toml` file and then restart the cluster.
```

File: reference/tiflash/tune-performance.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+    ```toml
+    [coprocessor]
+    region-max-size = "288MiB"
+    region-split-size = "192MiB"
+    region-max-keys = 2880000
+    region-split-keys = 1920000
+    ```
+
++ Add the following configuration in the `pd.toml` file to enable Region Merge:
```
Comment (ran-huang) [inline reference/tiflash/tune-performance.md:85]:
```suggestion
+ To enable Region Merge, add the following configuration in the `pd.toml` file:
```

**Comment by TomShawn** [comment]
@ran-huang PTAL again, thanks!

File: reference/tiflash/tune-performance.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+    ```sql
+    set @@tidb_opt_agg_push_down = 1;
+    ```
+
+## Configure the size of TiKV Regions
+
+The number and size of Regions might have some impact on TiFlash performance. Too many small Regions might bring down TiFlash performance by several times.
+
+For user scenarios that focus on analysis and query, it is recommended that you set the Region size to 192MB, and enable the Region Merge feature to ensure that small Regions generated during operation are automatically merged.
```
Comment (ran-huang) [inline reference/tiflash/tune-performance.md:37]:
```suggestion
For user scenarios that focus on analysis and query, it is recommended that you set the Region size to 192 MB, and enable the Region Merge feature to ensure that small Regions generated during operation are automatically merged.
```

**Comment by TomShawn** [comment]
@ilovesoup PTAL, thanks!

File: reference/tiflash/tune-performance.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+2. Enable the optimization for TiDB Operator such as the aggregate pushdown of `JOIN` or `UNION`:
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    set @@tidb_opt_agg_push_down = 1;
+    ```
+
+## Configure the size of TiKV Regions
```
Comment (ilovesoup) [inline reference/tiflash/tune-performance.md:33]: This section can be removed for now. I will pr for CN version as well. We now have an optimization for this case and region size will not be a problem in near future.

Comment (yikeke) [reply reference/tiflash/tune-performance.md:33]: See https://github.com/pingcap/docs-cn/pull/2686. @TomShawn

Comment (TomShawn) [reply reference/tiflash/tune-performance.md]: Removed.

File: reference/tiflash/tune-performance.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
+---
+title: Tune TiFlash Performance
+summary: Learn how to tune the performance of TiFlash.
+category: reference
+---
+
+# Tune TiFlash Performance
+
+This document introduces how to tune the performance of TiFlash, including planning machine resources and tuning TiDB parameters
```
Comment (yikeke) [inline reference/tiflash/tune-performance.md:9]:
```suggestion
This document introduces how to tune the performance of TiFlash, including planning machine resources and tuning TiDB parameters.
```

**Pull request merged**